src/Tribe/Modules/Single_Event_Template.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Tribe\Extensions\EA_Additional_Options\Modules;
+
+class Single_Event_Template {
+
+	public function hook() {
+		$template_setting     = tribe_get_option( Settings::PREFIX . 'default_template', false );
+		$block_editor_enabled = tribe_get_option( 'toggle_blocks_editor', null );
+
+		/** Make sure it only runs if:
+		 * 1. The block editor for events is enabled.
+		 * 2. A template is set up.
+		 */
+		if ( tribe_is_truthy( $block_editor_enabled ) && $template_setting ) {
+			add_filter( 'tribe_aggregator_event_translate_service_data_field_map', [ $this, 'adjust_field_map' ] );
+			add_filter( 'tribe_aggregator_before_save_event', [ $this, 'tec_ea_single_event_template' ], 11, 2 );
+		}
+	}
+
+	/**
+	 * Adjust the field map.
+	 *
+	 * @param array $fieldMap The field map from the source to the event data.
+	 *
+	 * @return array
+	 */
+	public function adjust_field_map( array $fieldMap ): array {
+		if ( isset( $fieldMap['description'] ) ) {
+			unset( $fieldMap['description'] );
+		}
+		$fieldMap['unsafe_description'] = 'post_content';
+		$fieldMap['start_date_utc']     = 'EventUTCStartDate';
+		$fieldMap['end_date_utc']       = 'EventUTCEndDate';
+
+		return $fieldMap;
+	}
+
+	/**
+	 * Create a block editor version of the imported event(s) based on the draft event defined below by post ID.
+	 *
+	 * @param object $event The event object being handled.
+	 * @param object $data  The import metadata object.
+	 *
+	 * @return object The event object.
+	 */
+	public function tec_ea_single_event_template( object $event, object $data ): object {
+
+		// Make sure this is only done on valid origins.
+		$valid_origins = [
+			'ical',
+			'gcal',
+			'ics',
+		];
+
+		/**
+		 * Allows filtering the accepted origins.
+		 * Useful if you want to limit it to certain type(s) of source(s).
+		 *
+		 * @var array $valid_origins The array of origins.
+		 */
+		$valid_origins = apply_filters( 'tribe_ext_ea_additional_options_valid_origins', $valid_origins );
+
+		if ( ! in_array( $data->meta['origin'], $valid_origins ) ) {
+			return $event;
+		}
+
+		$valid_sources = [];
+
+		/**
+		 * Allows filtering the source URLs to which the template should be applied to.
+		 * If you want to limit this to a certain source URL, set it here.
+		 *
+		 * @var array $valid_sources The array of the import URLs
+		 */
+		$valid_sources = apply_filters( 'tribe_ext_ea_additional_options_valid_sources', $valid_sources );
+
+		if (
+			! empty( $valid_sources )
+			&& ! in_array( $data->meta['source'], $valid_sources )
+		) {
+			return $event;
+		}
+
+		// Get the template from the saved setting.
+		$template_post_id = tribe_get_option( Settings::PREFIX . 'default_template', false );
+
+		if ( $template_post_id == 'enable' ) {
+			$blocks = [
```

Comment:
Make the `$blocks` value filterable to allow easy third-party customization without requiring modification of the original code.

Reply:
You know, I was just thinking about that.

Reply:
✔️ Added filtering.